core/src/test/java/com/amplifyframework/core/ShakeDetectorTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.core;
+
+import android.content.Context;
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+
+import com.amplifyframework.testutils.Resources;
+import com.amplifyframework.testutils.Sleep;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import io.reactivex.Completable;
+import io.reactivex.schedulers.Schedulers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests the {@link ShakeDetector} behavior.
+ */
+@RunWith(RobolectricTestRunner.class)
+public class ShakeDetectorTest {
+    // Amount of time in milliseconds between sending sensor
+    // events to the sensor listener.
+    private static final int SENSOR_DATA_DELAY = 65;
+    // Maximum amount of time in seconds before a test times out if the callback
+    // method has not been called.
+    private static final int TIMEOUT = 5;
+
+    /**
+     * Test that the callback method is called when a shake
+     * event is triggered.
+     */
+    @Test
+    public void shakeTriggersCallback() {
+        simulateSensorEvent("shake-accel-values.csv");
```

Comment:
I deliberated on whether or not you could use a [`ParameterizedRobolectricTestRunner`](http://robolectric.org/javadoc/4.3/org/robolectric/ParameterizedRobolectricTestRunner.html) for the test runner. Then, you could run the same test boy against different `.csv`. However, it is probably overkill. It is an interesting tool to be aware of, though. I've never actually used that one. Only `Parameterized` in backend development, and `RobolectricTestRunner` in front-end development; never the combo!